app/scripts/lib/buy-url.js

```diff
@@ -119,17 +119,20 @@ const createMoonPayUrl = async (walletAddress, chainId) => {
  *
  * @param {string} walletAddress - Ethereum destination address
  * @param {string} chainId - Current chain ID
+ * @param {string|undefined} symbol - Token symbol to buy
  * @returns String
  */
-const createCoinbasePayUrl = (walletAddress, chainId) => {
-  const { coinbasePayCurrencies } = BUYABLE_CHAINS_MAP[chainId];
+const createCoinbasePayUrl = (walletAddress, chainId, symbol) => {
+  // since coinbasePayCurrencies is going to be extended to include all tokens supported
+  // we now default to nativeCurrency instead of the 2 previous tokens + eth that we had before
+  const { nativeCurrency } = BUYABLE_CHAINS_MAP[chainId];
   const queryParams = new URLSearchParams({
     appId: COINBASEPAY_API_KEY,
     attribution: 'extension',
     destinationWallets: JSON.stringify([
       {
         address: walletAddress,
-        assets: coinbasePayCurrencies,
+        assets: symbol ? [symbol] : [nativeCurrency],
```

Comment:
Also with this change, the only purpose 'coinbasePayCurrencies' plays in the code base is as a check to see if coinbaseBayCurrencies exists and therefore coinbase is enabled for that chain. In that case I would suggest we add a new property to our settings that can be used for enabled checks instead of the existence of an array.

Reply:
the case with coinbase its a bit different, they need an array, so thats why we send an array, but the one item array allows us to skip a step in the purchase process

**before**:
only buying ETH was available as an action from the metamask side, but the 3 currency array was to enable the user to select another currency on a first step at the coinbase side

**now**:
since now multiple currencies can be selected on the metamask side, by sending a one item list, we skip the selection step on the coinbase side. this removes friction on the payment flow, since it removes a duplicate step which would be selecting the currency twice

the `coinbasePayCurrencies ` list, still has an important role, since just like transak, we use the list to check if that currency is available for purchase.